### PR TITLE
Caddy, security context and config

### LINKED ISSUE #161: Serve Frontend with Caddy
Let Caddy serve static files. Better performance and lower resource utilization than most other options. Also more or less a standard thing to do.

Copied from bcgov/nr-spar-webapp, which was archived.

https://github.com/bcgov/nr-spar-webapp/issues/48

### PR DESCRIPTION
**Pull request by DerekRoberts**
# Description

- Replace `react-serve` by `Caddy`
- Replace `react-scripts` and `webpack` by `Vite`
- Replace `jest` by `vitest`
- Remove `react-inject-env` from Docker and lets Caddy handle it
- Replace `yarn` by `npm`
- Bumps `jest` to 29.6.2 (**Need to check if we still need this package**)

About env vars:
> It looks like the frontend is using environment variables from build time, not run time.  We likely need to pick up those variables from a file, like `/env.js`, which is created by the Caddyfile.

Potentially usable examples:
https://github.com/bcgov/nr-fom/tree/main/public
https://github.com/bcgov/nr-compliance-enforcement/blob/bf9880d416cbdebafc4e470bd04ef784d0df81d7/frontend/src/config.js

Fixes #161, #54 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

These changes were extensively tested locally, with unit tests and in the deployed dev instance.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-284-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-284-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by RMCampos** [comment]
@craigyu and @DerekRoberts I would like to help with this change, since I did something similar on Sample-App. But we would need to add it to our sprint 😬

**Comment by DerekRoberts** [comment]
Content Security Policy seems to like blocking our javascript.  csper.io recommendings moving anything like that into its own file.
https://csper.io/docs/guide-removing-inline

**Comment by DerekRoberts** [comment]
<h5 _ngcontent-rnk-c8="" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0.5rem; font-weight: 500; line-height: 1.2; font-size: 1.25rem; color: rgb(33, 37, 41); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Inline Helper</h5><p _ngcontent-rnk-c8="" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 1rem; color: rgb(33, 37, 41); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">For content-security-policy to work correctly, there shouldn't be any inline resources. Here is all the inline resources found that need to be moved to their own file.<span> </span><a _ngcontent-rnk-c8="" href="https://csper.io/docs/guide-removing-inline" target="_blank" style="box-sizing: border-box; color: rgb(0, 123, 255); text-decoration: none; background-color: transparent;">More Info</a></p><div _ngcontent-rnk-c8="" style="box-sizing: border-box; color: rgb(33, 37, 41); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"></div><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"><div _ngcontent-rnk-c8="" style="box-sizing: border-box;"><h5 _ngcontent-rnk-c8="" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0.5rem; font-weight: 500; line-height: 1.2; font-size: 1.25rem;">script-src</h5><div _ngcontent-rnk-c8="" class="table-responsive" style="box-sizing: border-box; display: block; width: 794.444px; overflow-x: auto;">

Location | Line | Column | Script Sample
-- | -- | -- | --
https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/silent-check..https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/index-2ba20155.js | 153 | 40257 | parent.postMessage(location.href, locati

</div></div></div></div>Inline Helper
For content-security-policy to work correctly, there shouldn't be any inline resources. Here is all the inline resources found that need to be moved to their own file. [More Info](https://csper.io/docs/guide-removing-inline)

script-src
Location	Line	Column	Script Sample
[https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/silent-check..](https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/silent-check-sso)
https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/index-2ba20155.js	153	40257	parent.postMessage(location.href, locati
style-src
Location	Line	Column	Script Sample
[https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/empty..](https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/emptyUser-708ee209.jpg)
0	0	margin: 0px; background: #0e0e0e; height
[https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/empty..](https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/emptyUser-708ee209.jpg)
0	0	display: block;-webkit-user-select: none
[https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/empty..](https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/emptyUser-708ee209.jpg)
0	0	display: flex;flex-direction: column;ali
[https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/index..](https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/index-2ba20155.js)
https://nr-spar-284-frontend.apps.silver.devops.gov.bc.ca/assets/index-2ba20155.js	1	0

**Comment by RMCampos** [comment]
Found an issue on Carbon `HeaderContainer` and `SideNav` components. Issued a bug here: https://github.com/carbon-design-system/carbon/issues/14151

**Comment by RMCampos** [comment]
Blocked by new version of SASS: https://github.com/carbon-design-system/carbon/issues/14415

**Comment by DerekRoberts** [comment]
@RMCampos Good progress!  Shame about the block.  Want to close this and we revisit later?

File: frontend/Dockerfile [hunk 1]
```diff
@@ -1,26 +1,26 @@
-FROM node:18.17.0-alpine3.17 AS build
+# Build static files
+FROM node:18.16.1-alpine3.17 AS build
```
Comment (DerekRoberts) [inline frontend/Dockerfile:2]: This container is a step down from 18.17.0.  Please ignore if this was on purpose.

**Comment by DerekRoberts** [comment]
@RMCampos and I have decided to split this PR up.  Please keep watch for the follow-up.

**Pull request closed without merging**